### PR TITLE
CI update k8s and vault versions

### LINKED ISSUE #32: Update Kubernetes versions in test matrix
Update current versions to latest patch versions and add recent versions.

Reorganize versions in alphabetical order.

### PR DESCRIPTION
**Pull request by akijakya**
resolves #32

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -87,8 +87,8 @@ jobs:
     needs: [artifacts]
     strategy:
       matrix:
-        k8s_version: ["v1.25.8", "v1.24.12", "v1.23.17"]
-        vault_version: ["1.13.1", "1.12.5", "1.11.9", "1.10.11"]
+        k8s_version: ["v1.24.13", "v1.25.9", "v1.26.4" ]
```
Comment (sagikazarmark) [inline .github/workflows/ci.yaml:90]: I believe the latest kind release supports 1.27 as well.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -87,8 +87,8 @@ jobs:
...
+        vault_version: ["1.10.11", "1.11.10", "1.12.6", "1.13.2"] # https://hub.docker.com/_/vault
```
Comment (sagikazarmark) [inline .github/workflows/ci.yaml:91]:
```suggestion
        vault_version: ["1.10.11", "1.11.10", "1.12.6", "1.13.2"]
```

**Review by sagikazarmark — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (akijakya) [reply .github/workflows/ci.yaml:90]: Thats right, but afaik we agreed on that the in case of k8s, we support 3 versions until the latest-1 version.

Comment (sagikazarmark) [reply .github/workflows/ci.yaml:90]: I'm a but confused: why wouldn't we want to support the latest version? Unless of course there is some sort of incompatibility that keeps us from supporting it (and requires a breaking change to do so).

Comment (akijakya) [reply .github/workflows/ci.yaml:90]: I don't know, but I agree that it wouldn't hurt to add the latest version to this matrix 🙂 Should we drop 1.24 or let this be a 4 element list?

Comment (sagikazarmark) [reply .github/workflows/ci.yaml]: Personally, I'd keep as much versions as we can.

**Review by sagikazarmark — APPROVED** [review]

**Pull request merged**